docs/API.md

```diff
@@ -1565,6 +1565,75 @@ if err != nil {
 }
 ```
 
+<a name="SetBucketObjectLockConfig"></a>
+### SetBucketObjectLockConfig(bucketname, mode *RetentionMode, validity *uint, unit *ValidityUnit) error
+Set object lock configuration in given bucket. mode, validity and unit are either all set or all nil.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`bucketName` | _string_  |Name of the bucket|
+|`mode` | _RetentionMode_  |Retention mode to be set |
+|`validity` | _uint_  |Validity period to be set |
+|`unit` | _ValidityUnit_  |Unit of validity period |
+
+__Return Values__
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`err` | _error_  |Standard Error   |
+
+__Example__
+
+```go
+mode := Governance
+validity := uint(30)
+unit := Days
+
+err = minioClient.SetBucketObjectLockConfig("my-bucketname", &mode, &validity, &unit)
+if err != nil {
+    fmt.Println(err)
+    return
+}
+```
+
+<a name="GetBucketObjectLockConfig"></a>
+### GetBucketObjectLockConfig(bucketName) (*RetentionMode, *uint, *ValidityUnit, error)
+Get object lock configuration of given bucket.
+
+__Parameters__
+
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`bucketName`  | _string_  |Name of the bucket   |
+
+__Return Values__
+
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`mode` | _RetentionMode_  |Current retention mode |
+|`validity` | _uint_  |Current validity period |
+|`unit` | _ValidityUnit_  |Unit of validity period |
+|`err` | _error_  |Standard Error  |
+
+__Example__
+
+```go
+mode, validity, unit, err := minioClient.GetBucketLifecycle("my-bucketname")
```

Comment:
`GetBucketLifecycle` -> `GetBucketObjectLockConfig`
`minioClient`-> `s3Client` as it is an s3 feature till now.

Reply:
@balamurugana Can you please fix the typo here? Should we add versioning in `makeBucket` as we have done in `minio-java`?

Reply:
Fixed the doc.

No. it breaks backward compatibility.  I added 
```go
EnableVersioning(bucketName string) error
DisableVersioning(bucketName string) error
```